### PR TITLE
Add SFML

### PR DESCRIPTION
**Pull request by martinRenou**
Signed-off-by: martinRenou <martin.renou@gmail.com>

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml" 
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by martinRenou** [comment]
@conda-forge/help-c-cpp Everything but the OSX build is green,

I've got errors like:
```
ERROR :: $RPATH/../Frameworks/vorbis.framework/Versions/A/vorbis not in prefix_owners
```
for many dependencies like `vorbis`, even though `vorbis` is installed. Any clue what I should do to make it work?

Thanks!

**Comment by scopatz** [comment]
Hmm it seems like it is linking against the wrong vorbis, ie the system one and not the conda one.  Are there cmake variables you can set to fix the vorbis path?

**Comment by martinRenou** [comment]
The list of available cmake variables in SFML is here: https://www.sfml-dev.org/tutorials/2.5/compile-with-cmake.php

I did set `CMAKE_INSTALL_PREFIX` and `SFML_DEPENDENCIES_INSTALL_PREFIX` to the conda prefix.

**Comment by scopatz** [comment]
Sure, but there may be vorbis-specific variables that need to be set. That is pretty common with cmake

**Comment by martinRenou** [comment]
Ooooh, I think I see the problem. It looks like SFML vendors some external libs (and that is where cmake finds `vorbis`): https://github.com/SFML/SFML/tree/master/extlibs

What would be the "clean" way of dealing with this? I won't be legit for asking them to remove it from the SFML repository. The only thing I can think about would be to have a post-build removal of this `extlib` directory, would that be considered "clean"?

**Comment by martinRenou** [comment]
Don't mind what I said about the pre-build removal of the extlib directory. There seem to be a cmake option for that. It will try it.

**Comment by martinRenou** [comment]
Ok, so now the OSX build is green, but I am still concerned about it.

If finds OpenAL here: `/Applications/Xcode_10.1.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.14.sdk/System/Library/Frameworks/OpenAL.framework` even though I installed `openal-soft` with conda, so I am wondering if the Openal-soft recipe is not broken (not installing the framework? I am not really sure how that works).

Also, there are some ignored errors in the build:
```
  ERROR :: lib/libc++.1.dylib not in prefix_owners
  ERROR :: lib/libvorbisfile.3.dylib not in prefix_owners
  ERROR :: lib/libvorbis.0.dylib not in prefix_owners
  ERROR :: lib/libc++.1.dylib not in prefix_owners
  ERROR :: lib/libvorbisenc.2.dylib not in prefix_owners
  ERROR :: lib/libogg.0.dylib not in prefix_owners
  ERROR :: lib/libFLAC.8.dylib not in prefix_owners
  ERROR :: lib/libc++.1.dylib not in prefix_owners
  ERROR :: lib/libc++.1.dylib not in prefix_owners
  ERROR :: lib/libc++.1.dylib not in prefix_owners
  ERROR :: lib/libfreetype.6.dylib not in prefix_owners
```

**Comment by wolfv** [comment]
I am wondering if adding -DCMAKE_PREFIX_PATH=$PREFIX would help for the issue on OS X.

**Comment by martinRenou** [comment]
Thanks! I will try that

**Comment by wolfv** [comment]
also I couldn't see the build logs but on OS X there is a really annoying Mono.framework that has a couple of headers (for example the png header) which are _not_ the correct ones to pick up.

So the workaround is to add these lines:

https://github.com/conda-forge/sqlitecpp-feedstock/blob/384b70bd0d18ad46eefcdb5057cc5b2620cd5001/recipe/build.sh#L11-L18

but not sure if that's related to what you're seeing (can't judge without the build logs :)

**Comment by martinRenou** [comment]
The Remaining issue on Windows seems to be related to the libvorbis package on conda-forge. I might be wrong, but I feel like it's broken.

I opened a PR https://github.com/conda-forge/libvorbis-feedstock/pull/10

File: recipes/sfml/bld.bat [hunk 1]
```diff
@@ -0,0 +1,20 @@
+cmake -G "NMake Makefiles" ^
+  -D CMAKE_BUILD_TYPE=Release ^
+  -D CMAKE_INSTALL_PREFIX=%LIBRARY_PREFIX% ^
+  -D CMAKE_PREFIX_PATH=%LIBRARY_PREFIX% ^
```
Comment (xhochy) [inline recipes/sfml/bld.bat]: I'm wondering where the Windows equivalent of https://github.com/conda-forge/ctng-compiler-activation-feedstock/blob/50d5eca739d954e69798f0c4b3b56319a7922236/recipe/activate-gcc.sh#L98 is but this should also not be needed.

File: recipes/sfml/build.sh [hunk 2]
```diff
@@ -0,0 +1,22 @@
+#!/bin/bash
+
+cmake -D CMAKE_BUILD_TYPE=Release                 \
+      -D CMAKE_INSTALL_PREFIX=$PREFIX             \
+      -D UDEV_PATH_INCLUDES=$PREFIX/include       \
+      -D UDEV_PATH_LIB=$PREFIX/lib                \
+      -D CMAKE_INSTALL_LIBDIR=lib                 \
+      -D CMAKE_PREFIX_PATH=$PREFIX                \
```
Comment (xhochy) [inline recipes/sfml/build.sh]: This should not be needed as it is set by https://github.com/conda-forge/ctng-compiler-activation-feedstock/blob/50d5eca739d954e69798f0c4b3b56319a7922236/recipe/activate-gcc.sh#L98

File: recipes/sfml/bld.bat [hunk 3]
```diff
@@ -0,0 +1,20 @@
+cmake -G "NMake Makefiles" ^
```
Comment (xhochy) [inline recipes/sfml/bld.bat]: Can you use `Ninja` for parallel builds?

File: recipes/sfml/build.sh [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+      -D SFML_BUILD_FRAMEWORKS=OFF                \
+      -D SFML_BUILD_AUDIO=ON                      \
+      -D SFML_BUILD_GRAPHICS=ON                   \
+      -D SFML_BUILD_WINDOW=ON                     \
+      -D SFML_BUILD_NETWORK=ON                    \
+      -D SFML_USE_SYSTEM_DEPS=ON                  \
+      -D SFML_INSTALL_XCODE_TEMPLATES=OFF         \
+      $SRC_DIR
+
+make install
```
Comment (xhochy) [inline recipes/sfml/build.sh]: Can you do parallel builds here?

**Comment by martinRenou** [comment]
Thanks a lot for your review @xhochy! I'll make the relevant changes.

**Comment by wolfv** [comment]
hmm on windows it still doesn't seem to find FLAC :/

**Comment by martinRenou** [comment]
Yes. I opened an issue here https://github.com/conda-forge/libflac-feedstock/issues/7 I suspect it's very similar to the issue I had with vorbis, which I could fix apparently https://github.com/conda-forge/libvorbis-feedstock/pull/10 (it seems to correctly find Vorbis on Windows now)

**Comment by martinRenou** [comment]
Looks like it works now! Still @xhochy 's review to take into account. Will do that tonight.

**Comment by martinRenou** [comment]
Done! @xhochy would you mind giving it a last look?

File: recipes/sfml/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+    - xorg-libx11                    # [linux]
+    - xorg-libxrandr                 # [linux]
+    - libogg
+    - freetype
+    - libflac
+    - libvorbis
+    - openal-soft
+    - cmake
+    - ninja
+    - make                           # [unix]
```
Comment (xhochy) [inline recipes/sfml/meta.yaml]: Do you still need `make`?

Comment (martinRenou) [reply recipes/sfml/meta.yaml]: I'm not sure. Let's remove it and see

**Comment by martinRenou** [comment]
Done :)

**Comment by xhochy** [comment]
Thanks, merged!

**Comment by martinRenou** [comment]
Great! Thanks a lot for your reviews

**Pull request merged**